docs: Getting started guide (#577)
muxinc/media-chrome: docs/src/pages/en/get-started.mdx

layout: ../../layouts/MainLayout.astro
---

import CopyToClipboard from "../../components/PageContent/CopyToClipboard.astro";

<h2>What is Media Chrome?</h2>

> Media Chrome is a collection of fully customizable media player controls and components. You can use these components to build a media player from scratch that would be difficult to achieve with an out-of-the-box player.

Reviewer: ```suggestion
Media Chrome is a collection of fully customizable media player controls and components. You can use these components to build a media player from scratch.
```

## Key Features

- **Web Components:** Use with or without a JavaScript framework, compatible with: React, Angular, VueJS, Svelte, and more.
- **Styling:** Use simple CSS to style components
- **Theming:** Build custom themes that group together controls and style
- **Accessible:** All controls express their state in an accessible way
- **Unified API:** Compatible with standard `<video>` and `<audio>` elements and many other media elements

## Example usage

[placeholder: code embed example of a simple player using video element]

## Install

Media Chrome is packaged as an ES6 JavaScript module which is supported by all evergreen browsers and Node v12+.

### Option 1: Hosted

Load the module in the `<head>` of your HTML page. Note the `type="module"`, that's important.

<CopyToClipboard>
```html
<script type="module" src="https://cdn.jsdelivr.net/npm/media-chrome@0/+esm"></script>
```
</CopyToClipboard>

> Modules are always loaded asynchronously by the browser, so it's ok to load them in the head 👍

### Option 2: Bundled via npm

> Install via NPM (you could also use Yarn)

Reviewer 2: ```suggestion
Install via npm (you could also use Yarn or your favorite package manager)
```

<CopyToClipboard>
```bash
npm install media-chrome --save
```
</CopyToClipboard>

Import into your app.

<CopyToClipboard>
```js
import 'media-chrome';
```
</CopyToClipboard>

Importing the package like this will automatically register all of the available components in the browser. This means you can add a control like `<media-play-button></media-play-button>` to your page.

## Using in your HTML

> The `<media-controller>` component handles all of the communication between controls and the media being played. Wrap your media element with a `<media-controller>`, and add `slot="media"` to it.

Reviewer: ```suggestion
The `<media-controller>` component handles all of the communication between controls and the media being presented. To use it, wrap your media element with a `<media-controller>`, and add `slot="media"` to it.
```

  html={`<media-controller>
  <video
    slot="media"
    src="https://stream.mux.com/A3VXy02VoUinw01pwyomEO3bHnG4P32xzV7u1j1FSzjNg/low.mp4"
    playsinline
    muted
  ></video>
 <media-control-bar>
    <media-play-button></media-play-button>
    <media-mute-button></media-mute-button>
    <media-time-range></media-time-range>
    <media-fullscreen-button></media-fullscreen-button>
  </media-control-bar>
</media-controller>`}
/>

## Install

Media Chrome is packaged as an ES6 JavaScript module which is supported by all evergreen browsers and Node v12+.

### Option 1: Hosted

Load the module in the `<head>` of your HTML page. Note the `type="module"`, that's important.

<CopyToClipboard>
```html
<script type="module" src="https://cdn.jsdelivr.net/npm/media-chrome@0/+esm"></script>
```
</CopyToClipboard>

> Modules are always loaded asynchronously by the browser, so it's ok to load them in the head 👍

### Option 2: Bundled via npm

Install via npm (you could also use Yarn or your favorite package manager)

<CopyToClipboard>
```bash
npm install media-chrome --save
```
</CopyToClipboard>

Import into your app.

<CopyToClipboard>
```js
import 'media-chrome';
```
</CopyToClipboard>

> Importing the package like this will automatically register all of the available components in the browser. This means you can add a control like `<media-play-button>` to your page.

Reviewer 3: suggestion(non-blocking): Since this doesn't actually pull in everything (theme component & themes, extras, experimental), maybe:

```suggestion
Importing the package like this will automatically register all of the available of the core Media Chrome components in the browser. This means you can add a control like `<media-play-button>` to your page.
```

Author: I'll update this, your edit suggestion has a double "of the" in the sentence.

  ></video>
  <media-play-button></media-play-button>
</media-controller>
````
</CopyToClipboard>

> When using a control outside of a `<media-controller>` element it needs to be told which media controller it expects to be associated with. To do this you can set the `media-controller` attribute.

Reviewer: ```suggestion
When using a control outside of a `<media-controller>` wrapper element, the control needs to specify which media controller it should be associated with. To do this you can set the `media-controller` attribute.
```

<CopyToClipboard>
```html
<media-play-button mediacontroller="myController"></media-play-button>
```
</CopyToClipboard>

> > You could also set the mediaController property of the element with JavaScript

Reviewer: ```suggestion
> You could also set the `mediaController` property of the element with JavaScript
```

```
</CopyToClipboard>

> Unlike the video element we didn't specify a "slot". This just means that the control will be placed into the default slot, which is reserved for controls.

## Taking it further

That's all it takes to start building a custom media player. If you'd like to dive deeper into customization you can check out these resources

- [Styling](/en/styling)
- [Theming](/en/themes)

> ## Compatible media elements

Reviewer 2: I guess once we get started on the individual element pages, we can probably move this section into an overview page of that section and then update the link above to point there?

Author: Yeah this deserves to be a full section on it's own. We can replace with a link out when it's available.
